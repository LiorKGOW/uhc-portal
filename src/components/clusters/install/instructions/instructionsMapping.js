import links, { channels } from '../../../../common/installLinks';

const instructionsMapping = {
  aws: {
    cloudProvider: 'AWS',
    ipi: {
      title: 'Install OpenShift on AWS with installer-provisioned infrastructure',
      docURL: links.INSTALL_AWSIPI_DOCS_LANDING,
      channel: channels.STABLE,
    },
    upi: {
      title: 'Install OpenShift on AWS with user-provisioned infrastructure',
      docURL: links.INSTALL_AWSUPI_GETTING_STARTED,
      channel: channels.STABLE,
    },
  },
  gcp: {
    cloudProvider: 'GCP',
    ipi: {
      title: 'Install OpenShift on GCP with installer-provisioned infrastructure',
      docURL: links.INSTALL_GCPIPI_GETTING_STARTED,
      channel: channels.STABLE,
    },
    upi: {
      title: 'Install OpenShift on GCP with user-provisioned infrastructure',
      docURL: links.INSTALL_GCPUPI_GETTING_STARTED,
      rhcosLearnMoreURL: links.INSTALL_GCPUPI_RHCOS_LEARN_MORE,
      channel: channels.STABLE,
    },
  },
  azure: {
    cloudProvider: 'Azure',
    ipi: {
      title: 'Install OpenShift on Azure with installer-provisioned infrastructure',
      channel: channels.STABLE,
      docURL: links.INSTALL_AZUREIPI_GETTING_STARTED,
    },
    upi: {
      title: 'Install OpenShift on Azure with user-provisioned infrastructure',
      channel: channels.PRE_RELEASE,
      docURL: links.INSTALL_AZUREUPI_GETTING_STARTED,
    },
  },
  ibmz: {
    cloudProvider: 'IBM-Z',
    title: 'Install OpenShift on IBM Z with user-provisioned infrastructure',
    rhcosDownloadURL: links.DOWNLOAD_RHCOS_LATEST_IBMZ,
    rhcosLearnMoreURL: links.INSTALL_IBMZ_RHCOS_LEARN_MORE,
    docURL: links.INSTALL_IBMZ_GETTING_STARTED,
    channel: channels.IBMZ,
  },
  bareMetal: {
    cloudProvider: 'Bare Metal',
    title: 'Install OpenShift on Bare Metal with user-provisioned infrastructure',
    rhcosLearnMoreURL: links.INSTALL_BAREMETAL_RHCOS_LEARN_MORE,
    channel: channels.STABLE,
    docURL: links.INSTALL_BAREMETAL_GETTING_STARTED,
  },
  vmware: {
    cloudProvider: 'VMWare vSphere',
    title: 'Install OpenShift on vSphere with user-provisioned infrastructure',
    docURL: links.INSTALL_VSPHERE_GETTING_STARTED,
    rhcosLearnMoreURL: links.INSTALL_VSPHERE_RHCOS_LEARN_MORE,
    channel: channels.STABLE,
  },
  power: {
    cloudProvider: 'Power',
    title: 'Install OpenShift on Power with user-provisioned infrastructure',
    channel: channels.PPC,
    rhcosDownloadURL: links.DOWNLOAD_RHCOS_LATEST_PPC,
    showPreReleasePageLink: false,
    docURL: links.INSTALL_POWER_GETTING_STARTED,
  },
  openstack: {
    cloudProvider: 'Red Hat OpenStack Platform',
    ipi: {
      title: 'Install OpenShift on Red Hat OpenStack Platform with installer-provisioned infrastructure',
      channel: channels.STABLE,
      docURL: links.INSTALL_OSPIPI_GETTING_STARTED,
    },
    upi: {
      title: 'Install OpenShift on Red Hat OpenStack Platform with user-provisioned infrastructure',
      docURL: links.INSTALL_OSPUPI_GETTING_STARTED,
      rhcosLearnMoreURL: links.INSTALL_OSPUPI_RHCOS_LEARN_MORE,
      rhcosDownloadURL: links.DOWNLOAD_RHCOS_LATEST,
      channel: channels.STABLE,
    },
  },
  rhv: {
    cloudProvider: 'Red Hat Virutalization',
    title: 'Install OpenShift on Red Hat Virtualization with installer-provisioned infrastructure',
    docURL: links.INSTALL_RHV_GETTING_STARTED,
    showPreReleasePageLink: false,
    channel: channels.STABLE,
  },
};

export default instructionsMapping;
