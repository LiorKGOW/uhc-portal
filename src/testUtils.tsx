import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
