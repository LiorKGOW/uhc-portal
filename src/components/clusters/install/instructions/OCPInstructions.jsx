import React from 'react';
import PropTypes from 'prop-types';
import {
  Card,
  CardBody,
  Text,
} from '@patternfly/react-core';
import DownloadsAndPullSecretSection from './components/DownloadsAndPullSecretSection';
import SubscriptionAndSupport from './components/SubscriptionAndSupport';
import GetStarted from './components/GetStarted';
import Instruction from '../../../common/Instruction';
import Instructions from '../../../common/Instructions';
import TokenErrorAlert from './components/TokenErrorAlert';
import instructionsMapping from './instructionsMapping';
import { tools, channels } from '../../../../common/installLinks.mjs';

const OCPInstructions = (props) => {
  const {
    token,
    cloudProviderID,
    rhcos,
    installer = tools.X86INSTALLER,
    channel,
    docURL,
    customizations = '',
    showPreReleaseDocs,
    preReleasePageLink,
    isBMIPI,
    isUPI,
    showPreReleasePageLink,
  } = props;
  const pendoID = window.location.pathname;
  const getStartedTitleText = instructionsMapping[cloudProviderID]?.publicCloud
    ? `Follow the documentation to configure your ${instructionsMapping[cloudProviderID].cloudProvider} account and run the installer`
    : 'Follow the instructions to configure your environment and install your cluster';
  return (
    <>
      <Card>
        <CardBody>
          {token.error && (
            <>
              <TokenErrorAlert token={token} />
              <div className="pf-u-mb-lg" />
            </>
          )}
          <Instructions>
            <Instruction>
              <DownloadsAndPullSecretSection
                showPreReleaseDocs={showPreReleaseDocs}
                preReleasePageLink={preReleasePageLink}
                token={token}
                pendoID={pendoID}
                cloudProviderID={cloudProviderID}
                rhcos={rhcos}
                tool={installer}
                channel={channel}
                isBMIPI={isBMIPI}
                showPreReleasePageLink={showPreReleasePageLink}
              />
            </Instruction>
            <Instruction>
              <Text component="h2">{getStartedTitleText}</Text>
              <GetStarted
                docURL={docURL}
                pendoID={pendoID}
                cloudProviderID={cloudProviderID}
                customizations={customizations}
                isBMIPI={isBMIPI}
                isUPI={isUPI}
              />
            </Instruction>
            <Instruction>
              <SubscriptionAndSupport />
            </Instruction>
          </Instructions>
        </CardBody>
      </Card>
    </>
  );
};

OCPInstructions.propTypes = {
  token: PropTypes.object.isRequired,
  cloudProviderID: PropTypes.string.isRequired,
  rhcos: PropTypes.object,
  installer: PropTypes.oneOf(Object.values(tools)),
  channel: PropTypes.oneOf(Object.values(channels)).isRequired,
  docURL: PropTypes.string.isRequired,
  showPreReleaseDocs: PropTypes.bool,
  preReleasePageLink: PropTypes.string,
  customizations: PropTypes.string,
  isBMIPI: PropTypes.bool,
  isUPI: PropTypes.bool,
  showPreReleasePageLink: PropTypes.bool,
};

OCPInstructions.defaultProps = {
  isBMIPI: false,
  isUPI: false,
};

export default OCPInstructions;
