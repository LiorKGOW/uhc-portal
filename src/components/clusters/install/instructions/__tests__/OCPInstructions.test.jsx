import React from 'react';
import get from 'lodash/get';

import { checkAccessibility, render, screen } from '~/testUtils';

import { architectures } from '../../../../../common/installLinks.mjs';
import instructionsMapping from '../instructionsMapping';
import OCPInstructions from '../OCPInstructions';

const ocpOptions = {};
const providers = [];
Object.keys(instructionsMapping).forEach((value) => {
  const cloudProviderID = value;
  const mapping = instructionsMapping[value];
  const { cloudProvider, customizations } = mapping;
  if (mapping.installer) {
    // not (yet) divided into x86/arm and/or ipi/upi
    ocpOptions[cloudProvider] = { cloudProviderID, customizations, ...mapping };
    providers.push(cloudProvider);
  } else {
    Object.values(architectures).forEach((arch) => {
      const ipi = get(mapping, [arch, 'ipi'], null);
      const upi = get(mapping, [arch, 'upi'], null);
      if (ipi) {
        ocpOptions[`${cloudProvider}-${arch}-ipi`] = { cloudProviderID, customizations, ...ipi };
        providers.push(`${cloudProvider}-${arch}-ipi`);
      }
      if (upi) {
        ocpOptions[`${cloudProvider}-${arch}-upi`] = { cloudProviderID, isUPI: true, ...upi };
        providers.push(`${cloudProvider}-${arch}-upi`);
      }
    });
    // not (yet) divided into x86/arm
    const ipi = get(mapping, 'ipi', null);
    const upi = get(mapping, 'upi', null);
    if (ipi) {
      ocpOptions[`${cloudProvider}-ipi`] = { cloudProviderID, customizations, ...ipi };
      providers.push(`${cloudProvider}-ipi`);
    }
    if (upi) {
      ocpOptions[`${cloudProvider}-upi`] = { cloudProviderID, isUPI: true, ...upi };
      providers.push(`${cloudProvider}-upi`);
    }
  }
});
providers.sort();

describe('<OCPInstructions />', () => {
  it.each(providers)(
    'is accessible with %s',
    async (provider) => {
      const option = ocpOptions[provider];
      const { container } = render(<OCPInstructions {...option} token={{}} />);

      expect(
        await screen.findByText(
          'Select the options that apply to your cluster in the dialog and save.',
        ),
      ).toBeInTheDocument();

      await checkAccessibility(container);
    },
    20000,
  );
});
