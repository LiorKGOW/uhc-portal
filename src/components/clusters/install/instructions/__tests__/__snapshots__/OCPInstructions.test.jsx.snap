// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Every OCP instruction page should render:  ARM Bare Metal 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            cloudProviderID="ARM Bare Metal"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="ARM Bare Metal"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal-network-customizations.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  AWS 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            cloudProviderID="AWS"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="AWS"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_aws/installing-aws-customizations.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  AWS-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="AWS"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="AWS"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_aws/installing-aws-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_aws/installing-aws-account.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  AWS-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="AWS"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="AWS"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_aws/installing-aws-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_aws/installing-aws-user-infra.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Alibaba Cloud 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Alibaba Cloud"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Alibaba Cloud"
            customizations=""
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_alibaba/installing-alibaba-customizations.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Azure-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Azure"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Azure"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_azure/installing-azure-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_azure/installing-azure-default.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Azure-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="preRelease"
            cloudProviderID="Azure"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Azure"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_azure/installing-azure-customizations.html"
            docURL="https://github.com/openshift/installer/blob/master/docs/user/azure/install_upi.md"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Bare Metal-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Bare Metal"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Bare Metal"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal-network-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Bare Metal-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Bare Metal"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Bare Metal"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal-network-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_bare_metal/installing-bare-metal.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  GCP-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="GCP"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="GCP"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-account.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  GCP-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="GCP"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="GCP"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_gcp/installing-gcp-user-infra.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  IBM Cloud 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="IBM Cloud"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="IBM Cloud"
            customizations=""
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_ibm_cloud_public/preparing-to-install-on-ibm-cloud.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  IBM-Z 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="IBM-Z"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            preReleasePageLink="/install/ibmz/pre-release"
            token={Object {}}
            tool="s390x-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="IBM-Z"
            customizations=""
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_ibm_z/installing-ibm-z.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Platform Agnostic 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Platform Agnostic"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Platform Agnostic"
            customizations=""
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_platform_agnostic/installing-platform-agnostic.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Power 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Power"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            preReleasePageLink="/install/power/pre-release"
            token={Object {}}
            tool="ppc64le-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Power"
            customizations=""
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_ibm_power/installing-ibm-power.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Red Hat OpenStack Platform-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Red Hat OpenStack Platform"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Red Hat OpenStack Platform"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_openstack/installing-openstack-installer-custom.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_openstack/installing-openstack-installer-custom.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Red Hat OpenStack Platform-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Red Hat OpenStack Platform"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Red Hat OpenStack Platform"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_openstack/installing-openstack-installer-custom.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_openstack/installing-openstack-user.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Red Hat Virutalization-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Red Hat Virutalization"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Red Hat Virutalization"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_rhv/installing-rhv-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_rhv/installing-rhv-default.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  Red Hat Virutalization-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="Red Hat Virutalization"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="Red Hat Virutalization"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_rhv/installing-rhv-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_rhv/installing-rhv-user-infra.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  VMware vSphere-ipi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="VMware vSphere"
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="VMware vSphere"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;

exports[`Every OCP instruction page should render:  VMware vSphere-upi 1`] = `
<Fragment>
  <Card>
    <CardBody>
      <Instructions>
        <Instruction>
          <DownloadsAndPullSecretSection
            channel="stable"
            cloudProviderID="VMware vSphere"
            displayRHCOSSection={true}
            isBMIPI={false}
            pendoID="/"
            token={Object {}}
            tool="x86_64-openshift-install"
          />
        </Instruction>
        <Instruction>
          <Text
            component="h2"
          >
            Follow the instructions to configure your environment and install your cluster
          </Text>
          <GetStarted
            cloudProviderID="VMware vSphere"
            customizations="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere-installer-provisioned-customizations.html"
            docURL="https://docs.openshift.com/container-platform/latest/installing/installing_vsphere/installing-vsphere.html"
            isBMIPI={false}
            pendoID="/"
          />
        </Instruction>
        <Instruction>
          <SubscriptionAndSupport />
        </Instruction>
      </Instructions>
    </CardBody>
  </Card>
</Fragment>
`;
